### PR TITLE
Update to es6

### PR DESCRIPTION
**Pull request by jason-fox**
* Remove JSHint
* Add esLint using standard tamia presets
* Replace var with let/const
* Fix or disable eslint errors
* Remove jshint overrides

### CODE REVIEW

**Comment by jason-fox** [comment]
@fgalan - could this PR be reviewed please?

File: package.json [hunk 1]
```diff
@@ -37,11 +37,13 @@
   "devDependencies": {
     "clear-require": "~1.0.1",
     "coveralls": "~3.0.2",
-    "eslint-plugin-jasmine": "~2.10.1",
+    "eslint": "^5.8.0",
+    "eslint-plugin-mocha": "^5.2.0",
```
Comment (fgalan) [inline package.json:41]:
```suggestion
    "eslint-plugin-mocha": "~5.2.0",
```

File: package.json [hunk 2]
```diff
@@ -37,11 +37,13 @@
   "devDependencies": {
     "clear-require": "~1.0.1",
     "coveralls": "~3.0.2",
-    "eslint-plugin-jasmine": "~2.10.1",
+    "eslint": "^5.8.0",
```
Comment (fgalan) [inline package.json:40]:
```suggestion
    "eslint": "~5.8.0",
```

**Comment by fgalan** [comment]
> @fgalan - could this PR be reviewed please?

Reviewed. Some comments about packages.json to go...

File: package.json (hunk 1 above)
Comment (jason-fox) [reply package.json]: Fixed 0985bf9

File: package.json (hunk 2 above)
Comment (jason-fox) [reply package.json]: Fixed 7446954

**Pull request merged**